packages/financial-templates-lib/src/logger/DiscordTicketTransport.ts

```
@@ -0,0 +1,94 @@
+}
+
+// Type guard for log info object.
+export const isDiscordTicketInfo = (info: unknown): info is DiscordTicketInfo => {
+  if (!isDictionary(info)) return false;
+  return (
+    typeof info.message === "string" && typeof info.mrkdwn === "string" && typeof info.discordTicketChannel === "string"
+  );
+};
+
+export class DiscordTicketTransport extends Transport {
+  private readonly botToken: string;
+  private readonly channelIds: { [key: string]: string };
+
+  private client: Client;
+
+  constructor(winstonOpts: TransportOptions, { botToken, channelIds = {} }: Config) {
+    super(winstonOpts);
+    this.botToken = botToken;
+    this.channelIds = channelIds;
+
+    this.client = new Client({ intents: [GatewayIntentBits.Guilds] });
+  }
+
+  // Note: info must be any because that's what the base class uses.
+  async log(info: any, callback: () => void): Promise<void> {
+    // We only try sending if we have all expected parameters and a matching channel ID.
+    const canSend = isDiscordTicketInfo(info) && info.discordTicketChannel in this.channelIds;
+
+    if (canSend) {
+      try {
+        if (!this.client.isReady()) await this.login(); // Log in if not yet established the connection.
+
+        // Get and verify requested Discord channel to post.
+        const channelId = this.channelIds[info.discordTicketChannel];
+        const channel = await this.client.channels.fetch(channelId);
+        if (channel === null) throw new Error(`Discord channel ${channelId} not available!`);
+        if (!channel.isTextBased()) throw new Error(`Invalid type for Discord channel ${channelId}!`);
+
+        // Prepend the $ticket command and concatenate message title and content separated by newline.
+        const message = `$ticket ${info.message}\n${removeAnchorTextFromLinks(info.mrkdwn)}`;
+
+        // Send the message.
+        await channel.send(message);
+      } catch (error) {
+        console.error("Discord Ticket error", error);
```

Is this failing silently?
Should we alert other parts of the system and potentially re-trying the Discord bot message?

---

Great point! Since these messages are mission critical, I think a retry totally makes sense here with at least a re-login to reset the connection if necessary.

---

Agree, we should handle transport errors properly. As it concerns multiple existing transports, I'll create a separate PR for this